src/components/MoviesList/MoviesList.tsx

```diff
@@ -1,15 +1,29 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './MoviesList.scss';
 import { MovieCard } from '../MovieCard';
 
 interface Props {
   movies: Movie[];
+  filter: string;
 }
 
-export const MoviesList: React.FC<Props> = ({ movies }) => (
-  <div className="movies">
-    {movies.map(movie => (
-      <MovieCard key={movie.imdbId} movie={movie} />
-    ))}
-  </div>
-);
+export const MoviesList: React.FC<Props> = ({ movies, filter }) => {
+  const [visibleMovies, setVisibleMovies] = useState(([...movies]));
```

Comment:
no need to create a separate state for values that are dependent on another state value or prop
just create a const variable in this way `const visibleMovies = movies.filter()` - every time movies or filter change component will be re-rendered and the const value will be recalculated

Also, it's better to do it in `App.tsx`, and pass as a prop already filtered `visibleMovies`. The reason: single responsibility principle - `MoviesList` should just render a list and shouldn't know anything about filter